connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java

```diff
@@ -0,0 +1,267 @@
+                .typeName(ColumnTypeName.INT).typePrecision(10).typeScale(0).position(0)
+                .partitionIndex(partitionIndex1)
+                .comment("Integer column")
+                .nullable(true);
+        Integer partitionIndex2 = partitionColumns.indexOf("s");
+        if (partitionIndex2 == -1) partitionIndex2 = null;
+        ColumnInfo c2 = new ColumnInfo().name("s").typeText("STRING")
+                .typeJson("{\"type\": \"string\"}")
+                .typeName(ColumnTypeName.STRING).position(1)
+                .partitionIndex(partitionIndex2)
+                .comment("String column")
+                .nullable(true);
+        String location = new File(new File(dataDir, catalogName), tableName).getCanonicalPath();
+        if (format == DataSourceFormat.DELTA) {
+            // The Delta path can't be empty, need to set up here.
+            String partitionClause;
+            if (partitionColumns.isEmpty()) {
+                partitionClause = "";
+            } else {
+                partitionClause = String.format(
+                        " PARTITIONED BY (%s)",
+                        String.join(", ", partitionColumns));
+            }
+            session.sql(String.format("CREATE TABLE delta.`%s`(i INT, s STRING) USING delta", location) + partitionClause);
+        }
+        CreateTable createTableRequest = new CreateTable()
+                .name(tableName)
+                .catalogName(catalogName)
+                .schemaName(schemaName)
+                .columns(Arrays.asList(c1, c2))
+                .comment(COMMENT)
+                .storageLocation(location)
+                .tableType(TableType.EXTERNAL)
+                .dataSourceFormat(format);
+        tableOperations.createTable(createTableRequest);
+    }
+
+    @Before
+    public void setUp() {
+        super.setUp();
+        schemaOperations = new SdkSchemaOperations(createApiClient(serverConfig));
+        tableOperations = new SdkTableOperations(createApiClient(serverConfig));
+        cleanUp();
+    }
+
+    @Override
+    protected CatalogOperations createCatalogOperations(ServerConfig serverConfig) {
+        return new SdkCatalogOperations(createApiClient(serverConfig));
+    }
+
+    @Override
+    protected void cleanUp() {
+        deleteCatalog(CATALOG_NAME);
+        deleteCatalog(SPARK_CATALOG);
+        try {
+            JavaUtils.deleteRecursively(dataDir);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        super.cleanUp();
+    }
+
+    private void deleteCatalog(String catalogName) {
+        String schemaFullName = catalogName + "." + SCHEMA_NAME;
+        deleteTable(schemaFullName + "." + PARQUET_TABLE);
+        deleteTable(schemaFullName + "." + DELTA_TABLE);
+        try {
+            if (schemaOperations.getSchema(schemaFullName) != null) {
+                schemaOperations.deleteSchema(schemaFullName);
+            }
```

Comment:
we don't necessarily need to do get -> delete since we're catching the exception anyway, we can directly try to do the delete